pipeline.go

```diff
@@ -200,57 +229,84 @@ func NewPipelineDefinition(path string, env *PipelineEnvironment) (*PipelineDefi
 	}
 	d := &PipelineDefinition{}
 	err = yaml.Unmarshal(data, d)
-	// Add default meta to services and steps
-	for name, s := range d.Services {
-		s.Meta.Init()
-		d.Services[name] = s
-	}
-	for name, s := range d.Steps {
-		s.Meta.Init()
-		s.Meta.KeepAlive = KeepAliveNo
-		d.Steps[name] = s
-	}
 	// Update with specific meta if defined
-	for name, meta := range env.Services {
-		s, ok := d.Services[name]
-		if ok {
-			s.Meta = meta
-			d.Services[name] = s
-		} else if !meta.Ignore {
-			log.Printf("Metadata: unknown service '%s'", name)
-		}
-	}
 	for name, meta := range env.Steps {
 		s, ok := d.Steps[name]
 		if ok {
+			meta.Type = s.Meta.Type
 			s.Meta = meta
-			s.Meta.KeepAlive = KeepAliveNo
+			if meta.Type == ServiceTypeStep {
+				s.Meta.KeepAlive = KeepAliveNo
+			}
 			d.Steps[name] = s
 		} else if !meta.Ignore {
 			log.Printf("Metadata: unknown step '%s'", name)
 		}
 	}
+	// Initialize all metadata
```

Comment:
I find it a bit confusing that this happens after already accessing `d.Steps` in line 234 it's not really initialization then

Reply:
I tried to fix this issue in [0024fd5](https://github.com/ad-freiburg/gantry/pull/16/commits/0024fd5a572ef8447024826e26e5063c7b56327f)